clippy_lints/src/vec.rs

```diff
@@ -17,12 +18,21 @@ use rustc_session::impl_lint_pass;
 use rustc_span::{sym, DesugaringKind, Span};
 
 #[expect(clippy::module_name_repetitions)]
-#[derive(Clone)]
 pub struct UselessVec {
-    pub too_large_for_stack: u64,
-    pub msrv: Msrv,
-    pub span_to_lint_map: BTreeMap<Span, Option<(HirId, SuggestedType, String, Applicability)>>,
-    pub allow_in_test: bool,
+    too_large_for_stack: u64,
+    msrv: Msrv,
+    span_to_lint_map: BTreeMap<Span, Option<(HirId, SuggestedType, String, Applicability)>>,
+    allow_in_test: bool,
+}
+impl UselessVec {
+    pub fn new(conf: &'static Conf) -> Self {
+        Self {
+            too_large_for_stack: conf.too_large_for_stack,
+            msrv: conf.msrv.clone(),
```

Comment:
I assume it's because MSRV is sometimes mutated during linting when visiting `#[clippy::msrv]`, while other configurations are not

Reply:
Might be a nice use case for `Cow<'static, Msrv>` since it'll avoid clones if code never uses `#[clippy::msrv]` but not sure if that's overdoing it

Reply:
It was taken as `'static`, but it wasn't needed.

The config type should really be `Option<RustcVersion>`. I don't really know why `Msrv` exists as a type at all.

Reply:
If you mean we should always be using `Option<RustcVersion>` instead of `Msrv` (a stack of `RustcVersion`s), then I don't see how nested `#[clippy::msrv]` could work (or even a single `#[clippy::msrv]` with a clippy.toml that also defines an msrv).

If a lint pass updates and overwrites its `Option<RustcVersion>` because of a `#[clippy::msrv]` then surely it loses the previous, enclosing msrv?

Reply:
The type stored in the config should be `Option<...>`. The `Msrv` should just be `Vec<...>` rather than it's own type.

Reply:
Ah, I see. Yeah, having `Msrv` in the config doesn't make much sense. The type exists, because there are some convenience functions implemented on it. I thought this was nicer back then. Not married to it though, if you want to change this.

But let's not block this PR on this.